src/java.base/share/classes/java/util/zip/CheckedOutputStream.java

```diff
@@ -39,10 +39,13 @@
  * @since 1.1
  */
 public class CheckedOutputStream extends FilterOutputStream {
-    private Checksum cksum;
+    private final Checksum cksum;
 
     /**
-     * Creates an output stream with the specified Checksum.
+     * Creates an output stream with the specified Checksum. A null
+     * value to either {@code out} or {@code cksum} will cause
+     * a {@link NullPointerException} to be thrown from the
+     * {@code write} methods of this {@code CheckedOutputStream}.
```

Comment:
What is the reason for specifying the NPE in the method description rather than a throws?

Reply:
These 2 classes, the `CheckedInputStream` and the `CheckedOutputStream` are slightly different from the rest of the classes in this changeset. This javadoc here is for the constructor of the `CheckedInputStream`. The implementation of the constructor just blindly assigns the passed argument to an internal member field and doesn't do any null check on the passed arguments. So it isn't throwing any `NullPointerException` here. So adding a `@throws` here would be incorrect. In theory, we could just change the implementation of this constructor to throw a `NullPointerException` if either of these arguments were null, but that might mean a potential breakage of some weird usage of the CheckedInputStream. So I decided to add the `NullPointerException` detail to the constructor description.

Do you think we should instead do something like this for this class and the `CheckedOutputStream` class:

```diff
diff --git a/src/java.base/share/classes/java/util/zip/CheckedInputStream.java b/src/java.base/share/classes/java/util/zip/CheckedInputStream.java
index 10f72b416d1..76cba26986e 100644
--- a/src/java.base/share/classes/java/util/zip/CheckedInputStream.java
+++ b/src/java.base/share/classes/java/util/zip/CheckedInputStream.java
@@ -41,10 +41,7 @@ public class CheckedInputStream extends FilterInputStream {
     private final Checksum cksum;
 
     /**
-     * Creates an input stream using the specified Checksum. A null
-     * value to either {@code in} or {@code cksum} will cause
-     * a {@link NullPointerException} to be thrown from the
-     * {@code read} methods of this {@code CheckedInputStream}.
+     * Creates an input stream using the specified Checksum.
      * @param in the input stream
      * @param cksum the Checksum
      */
@@ -57,6 +54,8 @@ public CheckedInputStream(InputStream in, Checksum cksum) {
      * Reads a byte. Will block if no input is available.
      * @return the byte read, or -1 if the end of the stream is reached.
      * @throws    IOException if an I/O error has occurred
+     * @throws    NullPointerException if this {@code CheckedInputStream} was
+     *            constructed with a {@code null} value for {@code in} or {@code cksum}
      */
     public int read() throws IOException {
         int b = in.read();
@@ -75,7 +74,9 @@ public int read() throws IOException {
      * @param len the maximum number of bytes read
      * @return    the actual number of bytes read, or -1 if the end
      *            of the stream is reached.
-     * @throws     NullPointerException If {@code buf} is {@code null}.
+     * @throws     NullPointerException If {@code buf} is {@code null} or
+     *             if this {@code CheckedInputStream} was
+     *             constructed with a {@code null} value for {@code in} or {@code cksum}.
      * @throws     IndexOutOfBoundsException If {@code off} is negative,
      * {@code len} is negative, or {@code len} is greater than
      * {@code buf.length - off}


```

Reply:
Given that subclasses could set these fields lazily (however remote the case might be), do you think we should then not specify the `NullPointerException` for the read methods on these 2 classes. In which case I can exclude these 2 classes from this PR.

Reply:
It would be simpler to drop CheckedInputStream and CheckedOutputStream and deal with them separately. It's unfortunate that this current effort has run into this but that comes with touching some of the JDK 1.0/1.1 era classes.

Reply:
Done. I've updated the PR to undo the changes that were proposed for `CheckedOutputStream` and `CheckedInputStream`. I've also updated the CSR with the latest specification diff from this PR.